src/v/redpanda/admin/server.cc

```diff
@@ -2812,9 +2812,7 @@ admin_server::self_test_start_handler(std::unique_ptr<ss::http::request> req) {
                 } else if (test_type == "cloud") {
                     r.ctos.push_back(cluster::cloudcheck_opts::from_json(obj));
                 } else {
-                    throw ss::httpd::bad_param_exception(
-                      "Unknown self_test 'type', valid options are 'disk', "
-                      "'network', or 'cloud'.");
+                    vlog(adminlog.info, "Unknown self test type {}", test_type);
```

Comment:
Instead of ignoring it entirely, can we return this as a warning or error? A clear message indicating that the test isn't available on certain nodes seems pretty palatable, especially if an admin knows they're in the middle of an upgrade

Reply:
The advantage to logging and ignoring is that in the case that `rpk cluster self-test start` is invoked with no arguments, `rpk` will default to providing all of `disk`, `network`, and `cloud` as test types in the request.

By ignoring the error and not returning it, we can allow `disk` and `network` checks to proceed on older clusters which don't have the `cloud` check implemented.

IMO, this is the preferred behavior, `rpk cluster self-test start` should invoke as many tests as possible across a cluster, and not be restricted by the "minimum" node. However, doing a better job of communicating the ignored test on the `rpk` API side could definitely be an improvement.

Reply:
but if we added a REST API to list which tests were supported (or inferred from version information if that is available) then the entire flow could be customized in RPK and the low-level APIs could still correctly report errors.

Reply:
>By ignoring the error and not returning it, we can allow disk and network checks to proceed on older clusters which don't have the cloud check implemented.

Can we ignore the error and still return it? I was thinking we would still run the tests, but also indicate that some tests were skipped

Reply:
Actually looking at the return type, it looks like there isn't a good place to call this out, at least in this method. Not sure exactly where but I imagine we could maybe track skipped tests in the self test backend or something...

Reply:
I was thinking that we could expand the `self_test_request` struct to contain a vector of unknown test requests, and `push_back()` the "unknown" test type string here. 

The result of this would be shown later alongside the other results in `rpk self-test status`, with a generic "unknown test type: {}" message. However, I thought it might be confusing to see the mixed output for a given test in a cluster, since some nodes may produce the "unknown" message and others will have results.

But, it may be the least evil solution here, since it is a bit less of a silent failure.

The REST API that returns compatible tests for a given node is a good idea, but a bit more in depth than what I had in mind for the scope of this fix.

Let me know what you both think is best.